packages/auto-approve/src/utils-for-pr-checking.ts

```diff
@@ -413,6 +527,25 @@ export function runVersioningValidation(versions: Versions): boolean {
   return !isMajorVersionChanging(versions) && isMinorVersionUpgraded(versions);
 }
 
+/**
+ * Runs additional validation checks when a version is upgraded to ensure that the
+ * version is only upgraded, not downgraded, and that the major version is not bumped.
+ * If the PR could have a rev or sha (as in Docker or Go PRs), this checks that
+ * the minor version is bumped, and/or the rev or sha is upgraded
+ *
+ * @param versions the versions object returned from getVersions, getVersionsV2, and getJavaVersions
+ * @returns true if the version is only bumped a minor, not a major
+ */
+export function runVersioningValidationWithShaOrRev(
```

Comment:
The naming of this method does not help readability. A better common naming scheme IMO for something returning a true/false value would be `isValidSomething`.